### PR TITLE
Add PartialEq and PartialOrd for NonZeroInt to Int

### PR DESCRIPTION
**Pull request by Lucretiel**
Propose adding `PartialEq` and `PartialOrd` to the `NonZeroInt` types to compare to their inner types. Seems like a pretty obvious change, but I'm obviously open to any feedback or suggestions.

I'd also like to implement `PartialEq<NonZeroInt> for Int`, but I wasn't sure what the implications were for implementing traits on primitive types like that.

### CODE REVIEW

**Comment by Lucretiel** [comment]
These appear to be `tidy` failures; it's not clear to me what's actually going wrong or if my changes are causing them.

**Comment by Dylan-DPC** [comment]
@Lucretiel did you run tidy ?

**Comment by Lucretiel** [comment]
Huh, I thought I ran rustfmt already. Published fixes.

**Review by Amanieu — COMMENTED** [review]
> I'd also like to implement PartialEq<NonZeroInt> for Int, but I wasn't sure what the implications were for implementing traits on primitive types like that.

I think this is fine.

**Comment by JohnCSimon** [comment]
Ping from triage: 
@Lucretiel can you please post your status on this PR and address the merge conflicts? Thanks.

**Comment by joelpalmer** [comment]
Ping from Triage: Closing due to inactivity. Please re-open with any updates and conflicts resolved. Thank you for the PR. @Lucretiel

File: src/libcore/num/mod.rs [hunk 1]
```diff
@@ -110,6 +111,20 @@ assert_eq!(size_of::<Option<core::num::", stringify!($Ty), ">>(), size_of::<", s
                 }
             }
 
+            #[unstable(feature = "nonzero_cmp_to_int", issue = "70855")]
```
[the remark this answers is not in the document]
Comment (Lucretiel) [reply src/libcore/num/mod.rs:114]: I updated to stable; it wants a "since" attribute

**Comment by Lucretiel** [comment]
Updated this branch! Conflicts resolved, tidy passing. Not sure how to account for making the trait insta-stable, I'll need some guidance on that.

**Comment by Amanieu** [comment]
You'll need to reopen the PR or open a new one.

For the stability attribute just mark it as since 1.45.0. We'll need to run this through FCP for stabilization.

**Comment by Lucretiel** [comment]
I don't have a "reopen" button available to me; maybe @joelpalmer?

**Comment by Lucretiel** [comment]
Opened #72405 instead

**Pull request closed without merging**